tensorboard/webapp/plugins/text/text_v2_plugin.py

```diff
@@ -0,0 +1,215 @@
+
+    Raises:
+      ValueError: If the argument is not a numpy ndarray, or the dimensionality
+        is too low.
+    """
+    if not isinstance(arr, np.ndarray):
+        raise ValueError("reduce_to_2d requires a numpy.ndarray")
+
+    ndims = len(arr.shape)
+    if ndims < 2:
+        raise ValueError("reduce_to_2d requires an array of dimensionality >=2")
+    # slice(None) is equivalent to `:`, so we take arr[0,0,...0,:,:]
+    slices = ([0] * (ndims - 2)) + [slice(None), slice(None)]
+    return arr[slices]
+
+
+def reduce_and_jsonify(text_arr):
+    """Take a numpy.ndarray containing strings, and convert it into a
+    json-compatible list, also squashing it to two dimensions if necessary.
+
+    If the ndarray contains a single scalar string, then that ndarray is
+    converted to a list. If it contains an array of strings,
+    that array is converted to a list.  If the array contains dimensionality
+    greater than 2, all but two of the dimensions are removed, and a warning
+    message is generated.  The list, shape of the array, and any warning
+    message is returned.
+
+    Args:
+      text_arr: A numpy.ndarray containing strings.
+
+    Returns:
+        The JSON-compatible list
+        The shape of the array
+        A warning message (empty string if no warning)
+    """
+    warning = ""
+    if not text_arr.shape:
+        # It is a scalar. No need to put it in a table, just apply markdown
+        return text_arr.tolist(), text_arr.shape, warning
+    if len(text_arr.shape) > 2:
+        warning = plugin_util.markdown_to_safe_html(
+            WARNING_TEMPLATE % len(text_arr.shape)
+        )
+        text_arr = reduce_to_2d(text_arr)
+    return text_arr.tolist(), text_arr.shape, warning
+
+
+def create_event(wall_time, step, string_ndarray):
+    """Convert a text event into a JSON-compatible response with rank <= 2"""
+    formatted_string_array, shape, warning = reduce_and_jsonify(string_ndarray)
+    return {
+        "wall_time": wall_time,
+        "step": step,
+        "text": formatted_string_array,
+        "shape": shape,
+        "warning": warning,
+    }
+
+
+class TextV2Plugin(base_plugin.TBPlugin):
+    """Angular Text Plugin For Tensorboard"""
+
+    plugin_name = "text_v2"
+
+    def __init__(self, context):
+        """Instantiates Angular TextPlugin via TensorBoard core.
+
+        Args:
+          context: A base_plugin.TBContext instance.
+        """
+        self._multiplexer = context.multiplexer
+        self._downsample_to = (context.sampling_hints or {}).get(
+            self.plugin_name, _DEFAULT_DOWNSAMPLING
+        )
+        if context.flags and context.flags.generic_data != "false":
+            self._data_provider = context.data_provider
+        else:
+            self._data_provider = None
+
+    def frontend_metadata(self):
+        return base_plugin.FrontendMetadata(
+            is_ng_component=True, tab_name="Text v2", disable_reload=False
```

Comment:
`disable_reload=False`, there is no reason to disable the reload (unlike the debugger). Enable it :)

Reply:
`disable_reload=False` ======> reload enabled or disabled?  Semantics are a bit confusing here

Reply:
resolved